modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs

```diff
@@ -202,6 +204,66 @@ public Error OpenInExternalEditor(Script script, int line, int col)
                 case ExternalEditorId.None:
                     // Not an error. Tells the caller to fallback to the global external editor settings or the built-in editor.
                     return Error.Unavailable;
+                case ExternalEditorId.CustomEditor:
+                {
+                    string file = ProjectSettings.GlobalizePath(script.ResourcePath);
+                    var execCommand = _editorSettings.GetSetting(Settings.CustomExecPath).As<string>();
+                    var execArgs = _editorSettings.GetSetting(Settings.CustomExecPathArgs).As<string>();
+                    var args = new List<string>();
+                    var from = 0;
+                    var numChars = 0;
+                    var insideQuotes = false;
+                    var hasFileFlag = false;
+
+                    execArgs = execArgs.ReplaceN("{line}", line.ToString());
+                    execArgs = execArgs.ReplaceN("{col}", col.ToString());
+                    execArgs = execArgs.StripEdges(true, true);
+                    execArgs = execArgs.Replace("\\\\", "\\");
+
+                    for (int i = 0; i < execArgs.Length; ++i)
+                    {
+                        if ((execArgs[i] == '"' && (i == 0 || execArgs[i - 1] != '\\')) && i != execArgs.Length - 1)
+                        {
+                            if (!insideQuotes)
+                            {
+                                from++;
+                            }
+                            insideQuotes = !insideQuotes;
+                        }
+                        else if ((execArgs[i] == ' ' && !insideQuotes) || i == execArgs.Length - 1)
+                        {
+                            if (i == execArgs.Length - 1 && !insideQuotes)
+                            {
+                                numChars++;
+                            }
+
+                            var arg = execArgs.Substr(from, numChars);
+                            if (arg.Contains("{file}"))
+                            {
+                                hasFileFlag = true;
+                            }
+
+                            arg = arg.ReplaceN("{file}", file);
```

Comment:
Tho this change is kinda hacky, so we may just no support this for the C# editor (esp, since here the solution path would probably be more useful):

```suggestion
                            arg = arg.ReplaceN("{project}", project_path);
                            arg = arg.ReplaceN("{file}", file);
```

and with the other variable:
```csharp
var project_path = ProjectSettings.GlobalizePath("res://");
if (project_path.EndsWith('/')) project_path = project_path[..^1];
```